src/controller/java/generated/java/matter/devicecontroller/cluster/clusters/AccessControlCluster.kt

```diff
@@ -37,144 +44,542 @@ class AccessControlCluster(
   class AttributeListAttribute(val value: List<UInt>)
 
   suspend fun readAclAttribute(): AclAttribute {
-    // Implementation needs to be added here
-  }
+    val ATTRIBUTE_ID_ACL: UInt = 0u
+
+    val attributePath =
+      AttributePath(endpointId = endpointId, clusterId = CLUSTER_ID, attributeId = ATTRIBUTE_ID_ACL)
+
+    val readRequest = ReadRequest(eventPaths = emptyList(), attributePaths = listOf(attributePath))
+
+    val response = controller.read(readRequest)
+
+    if (response.successes.isNotEmpty()) {
+      logger.log(Level.INFO, "Read command succeeded")
+
+      val attributeData =
+        response.successes.filterIsInstance<ReadData.Attribute>().firstOrNull {
+          it.path.attributeId == ATTRIBUTE_ID_ACL
+        }
+
+      requireNotNull(attributeData) { "Acl attribute not found in response" }
 
-  suspend fun readAclAttributeWithFabricFilter(isFabricFiltered: Boolean): AclAttribute {
-    // Implementation needs to be added here
+      // Decode the TLV data into the appropriate type
+      val tlvReader = TlvReader(attributeData.data)
+      val decodedValue: List<AccessControlClusterAccessControlEntryStruct> =
+        buildList<AccessControlClusterAccessControlEntryStruct> {
+          tlvReader.enterArray(AnonymousTag)
+          while (!tlvReader.isEndOfContainer()) {
+            add(AccessControlClusterAccessControlEntryStruct.fromTlv(AnonymousTag, tlvReader))
+          }
+          tlvReader.exitContainer()
+        }
+
+      return AclAttribute(decodedValue)
+    } else {
+      logger.log(Level.WARNING, "Read command failed")
+      throw IllegalStateException("Read command failed with failures: ${response.failures}")
+    }
   }
 
   suspend fun writeAclAttribute(
     value: List<AccessControlClusterAccessControlEntryStruct>,
     timedWriteTimeoutMs: Int? = null
   ) {
-    if (timedWriteTimeoutMs != null) {
-      // Do the action with timedWriteTimeoutMs
-    } else {
-      // Do the action without timedWriteTimeoutMs
+    val ATTRIBUTE_ID_ACL: UInt = 0u
+    val timeoutMs: Duration =
```

Comment:
maybe not in this PR, however would it make sense for the API to support timeout null instead of requiring ZERO for no timeout? duration.zero seems to indicate "must succeed immediately without a wait". It seems more intuitive to have null for missing.

Reply:
Yes, we need to update the Core API https://github.com/project-chip/connectedhomeip/issues/30901